src/olympia/amo/urlresolvers.py

```diff
@@ -36,8 +36,15 @@ def split_path(path_):
         first_lower = first.lower()
         lang, dash, territory = first_lower.partition('-')
 
-        # Check language-territory first.
-        if first_lower in settings.LANGUAGE_URL_MAP:
+        # First test shorter languages shortcuts.
+        if not dash and first in settings.SHORTER_LANGUAGES:
+            first = settings.SHORTER_LANGUAGES[first]
```

Comment:
Nit: the "first" renaming is blowing up my brain. I know what it's doing but it feels confusing. could you use a different name "first_short" for example?

Reply:
c25a40594bb1a88268eb63a7b7ba347c9cdd6e2b